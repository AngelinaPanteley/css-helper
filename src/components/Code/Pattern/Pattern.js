import React, { PureComponent } from 'react';
import { CopyToClipboard } from 'react-copy-to-clipboard';
import styles from './Pattern.scss';
import Icon from '../../UI/Icon/Icon';

class Pattern extends PureComponent {
  render() {
    return (
      <div className={styles.Wrapper}>
        <p className={styles.Title}>
          <span>{this.props.title}</span>
          <CopyToClipboard text={this.props.code}>
            <button className={styles.Button}
              onClick={this.props.onCopy}>
              <Icon icon="copy"></Icon>
            </button>
          </CopyToClipboard>
        </p>
        <pre className={styles.Pre}>
          <code className={styles.Code}>
            {this.props.code}
          </code >
        </pre>
      </div>
    )
  }
}

export default Pattern;