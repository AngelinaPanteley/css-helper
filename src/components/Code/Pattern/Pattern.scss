.Pre {
  width: 100%;
  overflow: auto;
  background: #0d0e10;
  padding: 10px;
  margin: 20px auto;
}

.Code {
  display: block;
  color: #b9b9bb;
  width: 100%;
}

.Title {
  color: #fff;
  font-size: 20px;
}

.Wrapper {
  margin-bottom: 60px;
}

.Button {
  background: transparent;
  border: none;
  color: #fff;
  font-size: 20px;
  cursor: pointer;
  outline: none;
  float: right;

  &:hover {
    text-shadow: 0 0 1px #fff;
    transition: all .1s ease-in-out;
  }
}