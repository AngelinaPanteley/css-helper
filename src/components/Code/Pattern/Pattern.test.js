"use strict";

import React from 'react';
import renderer from 'react-test-renderer';
import { expect as chaiExpect } from 'chai';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16.1';
import Pattern from './Pattern.js';

configure({ adapter: new Adapter() });

describe('>>>Pattern', () => {

  //SNAPSHOTS

  const props = {
    title: 'title',
    code: 'code',
    onCopy: () => { },
  };

  const patternComponent = <Pattern {...props} />;

  const component = renderer.create(patternComponent);

  it('renders pattern correctly', () => {
    let componentTree = component.toJSON();
    expect(componentTree).toMatchSnapshot();
  });

  //SHALLOW RENDERING

  let wrapper;

  beforeEach(() => {
    wrapper = shallow(patternComponent);
  });

  it('renders code', () => {
    chaiExpect(wrapper.contains(
      <pre>
        <code>
          {props.code}
        </code >
      </pre>)).to.equal(true);
  });

  it('renders title', () => {
    chaiExpect(wrapper.contains(
      <span>{props.title}</span>
    )).to.equal(true);
  });
})