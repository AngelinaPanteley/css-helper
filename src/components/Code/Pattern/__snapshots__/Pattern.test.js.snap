// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>Pattern renders pattern correctly 1`] = `
<div
  className={undefined}
>
  <p
    className={undefined}
  >
    <span>
      title
    </span>
    <button
      className={undefined}
      onClick={[Function]}
    >
      <i
        className="undefined undefined"
      />
    </button>
  </p>
  <pre
    className={undefined}
  >
    <code
      className={undefined}
    >
      code
    </code>
  </pre>
</div>
`;
